components/courses/app/policies/courses/season/graduate_students_policy.rb

```diff
@@ -0,0 +1,21 @@
+module Courses
+  class Season < ApplicationRecord
+    class GraduateStudentsPolicy
+      def initialize(season)
+        @season  = season
+      end
+
+      def allowed?
+        season.passed? && has_attending_students?
+      end
+
+      private
+
+      attr_reader :season
+
+      def has_attending_students?
+        season.students.where(status: :attending).any?
```

Comment:
`where(status: :attending)` is appearing in 2 places. You can use scope instead.